boards/ambiq/apollo510_evb/apollo510_evb.dts

```diff
@@ -99,6 +104,85 @@
 	status = "okay";
 };
 
+&mspi0 {
+	pinctrl-0 = <&mspi0_default>;
+	pinctrl-1 = <&mspi0_sleep>;
+	pinctrl-2 = <&mspi0_psram>;
+	pinctrl-names = "default","sleep","psram";
+	status = "okay";
+	zephyr,pm-device-runtime-auto;
+
+	ce-gpios = <&gpio192_223 7 GPIO_ACTIVE_LOW>;
+
+	cmdq-buffer-location = "SRAM_NO_CACHE";
+	cmdq-buffer-size = <256>;
+
+	ambiq,apmemory;
+
+	dqs-support;
+
+	psram0: aps51216ba@0 {
+		compatible = "ambiq,mspi-device", "mspi-aps-z8";
```

Comment:
Second binding already included mspi-device so you should be doing the below? (and in any case, compatibles should be ordered from most specific to least specific since it stops at the first match)

```suggestion
		compatible = "mspi-aps-z8";
```